core/src/main/java/org/stellar/anchor/client/DefaultClientService.java

```diff
@@ -0,0 +1,86 @@
+package org.stellar.anchor.client;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.io.FilenameUtils;
+import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.api.exception.SepNotFoundException;
+import org.stellar.anchor.config.ClientsConfig;
+import org.stellar.anchor.util.FileUtil;
+import org.stellar.anchor.util.GsonUtils;
+import org.yaml.snakeyaml.Yaml;
+
+public class DefaultClientService implements ClientService {
+  static final Gson gson = GsonUtils.getInstance();
+  List<CustodialClientConfig> custodialClients;
+  List<NonCustodialClientConfig> nonCustodialClients;
+
+  public static DefaultClientService fromClientsConfig(ClientsConfig clientsConfig)
+      throws InvalidConfigException {
+    return switch (clientsConfig.getType()) {
+      case FILE -> fromFile(clientsConfig.getValue());
+      case URL -> throw new InvalidConfigException("URL is not supported yet.");
+      default -> throw new InvalidConfigException(
+          String.format("clients.type:%s is not supported.", clientsConfig.getType()));
+    };
+  }
+
+  public static DefaultClientService fromFile(String filePath) throws InvalidConfigException {
+    Map<String, Object> map = parseFileToMap(filePath);
+    return createDCSFromMap(map);
+  }
+
+  private static Map<String, Object> parseFileToMap(String filePath) throws InvalidConfigException {
+    try {
+      String fileContent = FileUtil.read(Path.of(filePath));
+      return switch (FilenameUtils.getExtension(filePath).toLowerCase()) {
+        case "json" -> gson.fromJson(fileContent, Map.class);
+        case "yaml", "yml" -> new Yaml().load(fileContent);
+        default -> throw new InvalidConfigException(
+            String.format("%s is not a supported file format", filePath));
+      };
+    } catch (Exception ex) {
+      throw new InvalidConfigException(
+          List.of(String.format("Cannot read from clients file: %s", filePath)), ex);
+    }
+  }
+
+  private static DefaultClientService createDCSFromMap(Map<String, Object> map) {
+    DefaultClientService clientService = new DefaultClientService();
+    clientService.custodialClients =
+        gson.fromJson(gson.toJson(map.get("custodial_clients")), List.class);
+    clientService.nonCustodialClients =
+        gson.fromJson(gson.toJson(map.get("noncustodial_clients")), List.class);
+    return clientService;
+  }
```

Comment:
So is the file expected to have `custodial_clients` and `noncustodial_clients` attributes? This is not the formal of our current clients config. Why are we changing the format? If we want to use a different interface internally that is fine, but I don't see a reason to change the file schema

Reply:
The new format is more clear in terms of specifying which client should have what fields defined and is easier to validate. Since the ClientsConfig is new and I don't think businesses have started using it yet, this is an opportunity to establish the right configuration from the outset.

Reply:
Ok, I don't feel too strongly about this, but can we update format to avoid 2 top-level keys?

Maybe:

```yaml
clients:
  noncustodial:
    - ...
  custodial:
    - ...
```